frontend/isoko/src/features/dashboard/DashboardSlice.ts

```diff
@@ -0,0 +1,60 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Business } from '../../types/GlobalTypes';
+import { fetchBusinessDetails, updateBusinessDetails } from './DashboardAPI';
+
+export interface DashboardState {
+   business: Business | null;
+   analytics: null;
+   status: 'idle' | 'loading' | 'failed';
+}
+
+const initialState: DashboardState = {
+   business: null,
+   analytics: null,
+   status: 'idle',
+};
+
+export const initializeBusinessDetailsAsync = createAsyncThunk(
+   'dashboard/initializeBusinessDetails',
+   async (businessId: string) => {
+      return await fetchBusinessDetails(businessId);
+   }
+);
+
+export interface UpdateParams {
+   businessId: string;
+   [key: string]: any;
```

Comment:
Does this allow for more than 1 business detail to be updated?

Reply:
Okay it seems like it does based on the test. Then I guess a general question is does that structure [key: string]: any; allow you to add any number of fields with that format?

Reply:
Yes, and yes. This type basically says accept any object that has businessId: string and accept anything for the rest of the object.